src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2303,6 +2303,70 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         tmp = nullptr;
     }
 
+    ASR::asr_t* handle_intrinsic_int(Allocator &al, Vec<ASR::call_arg_t> args,
+                                        const Location &loc) {
+        ASR::expr_t *arg = nullptr, *value = nullptr;
+        ASR::ttype_t *type = nullptr;
+        if (args.size() > 0) {
+            arg = args[0].m_value;
+            type = ASRUtils::expr_type(arg);
+        }
+        ASR::ttype_t *to_type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
+                                    4, nullptr, 0));
+        if (!arg) {
+            return ASR::make_ConstantInteger_t(al, loc, 0, to_type);
+        }
+        if (ASRUtils::is_real(*type)) {
+            if (ASRUtils::expr_value(arg) != nullptr) {
+                int32_t ival = ASR::down_cast<ASR::ConstantReal_t>(
+                                        ASRUtils::expr_value(arg))->m_r;
+                value =  ASR::down_cast<ASR::expr_t>(make_ConstantInteger_t(al,
+                                loc, ival, to_type));
```

Comment:
So right now `int` always returns 32bit integer. So if you do:
```
i: i64
i = int(3.0)
```
It will fail, because `int` returns a 32 bit integer, but `i` is a 64bit.

The way we do this in LPython is that on assignment `a = b`, we require both `a` and `b` to be exactly the same type (int/float/complex/...), but they don't have to be the same kind. Rather, LPython will insert implicit cast to cast from 64bit to 32bit and vice versa. 

For now I think int() should return a 64 bit integer, and then LPython implicitly casts it to 32bit if needed. The same with float(), which I think should return a 64bit float, and LPython casts it to 32bit float if needd.

This can be implemented in a future PR.

Reply:
Sure, will do that.